client/src/hooks/useFeatureFlag.js

```diff
@@ -1,10 +1,17 @@
 import { useUserContext } from "../contexts/userContext";
+import { useFeatures } from "../hooks/useFeatures";
 
 export default function useFeatureFlag(flagName) {
   const { user } = useUserContext();
-  if (!user || !user.features) {
-    return false;
+  const { data: featureFlags } = useFeatures();
+  const featureFlag = featureFlags.find((feature) => feature.name === flagName);
+
+  if (featureFlag && featureFlag.is_enabled) {
+    return true;
+  }
+  if (user && user.features && user.features.includes(flagName)) {
+    return true;
```

Comment:
I think we can move these conditions to a variable so it's easier to read and follow like
```
const isFeatureEnabled = ...
const userHasFeature = ...
```
so we can just check `if (isFeatureEnabled || userHasFeature)`